data_model/src/isi.rs

```diff
@@ -79,790 +71,1151 @@ pub mod model {
         #[debug(fmt = "{_0:?}")]
-        NewParameter(NewParameterExpr),
-        Upgrade(UpgradeExpr),
-        /// `Log` variant.
-        Log(LogExpr),
+        Log(Log),
 
         #[debug(fmt = "{_0:?}")]
         Fail(Fail),
     }
 
-    impl Instruction for InstructionExpr {}
-
-    impl Instruction for SetKeyValueExpr {}
-    impl Instruction for RemoveKeyValueExpr {}
-    impl Instruction for RegisterExpr {}
-    impl Instruction for UnregisterExpr {}
-    impl Instruction for MintExpr {}
-    impl Instruction for BurnExpr {}
-    impl Instruction for TransferExpr {}
-    impl Instruction for GrantExpr {}
-    impl Instruction for RevokeExpr {}
-    impl Instruction for SetParameterExpr {}
-    impl Instruction for NewParameterExpr {}
-    impl Instruction for UpgradeExpr {}
-    impl Instruction for ExecuteTriggerExpr {}
-    impl Instruction for LogExpr {}
+    impl Instruction for InstructionBox {}
+
+    impl Instruction for SetKeyValue<Domain> {}
+    impl Instruction for SetKeyValue<Account> {}
+    impl Instruction for SetKeyValue<AssetDefinition> {}
+    impl Instruction for SetKeyValue<Asset> {}
+
+    impl Instruction for RemoveKeyValue<Domain> {}
+    impl Instruction for RemoveKeyValue<Account> {}
+    impl Instruction for RemoveKeyValue<AssetDefinition> {}
+    impl Instruction for RemoveKeyValue<Asset> {}
+
+    impl Instruction for Register<Peer> {}
+    impl Instruction for Register<Domain> {}
+    impl Instruction for Register<Account> {}
+    impl Instruction for Register<AssetDefinition> {}
+    impl Instruction for Register<Asset> {}
+    impl Instruction for Register<Role> {}
+    impl Instruction for Register<Trigger<TriggeringFilterBox>> {}
+
+    impl Instruction for Unregister<Peer> {}
+    impl Instruction for Unregister<Domain> {}
+    impl Instruction for Unregister<Account> {}
+    impl Instruction for Unregister<AssetDefinition> {}
+    impl Instruction for Unregister<Asset> {}
+    impl Instruction for Unregister<Role> {}
+    impl Instruction for Unregister<Trigger<TriggeringFilterBox>> {}
+
+    impl Instruction for Mint<PublicKey, Account> {}
+    impl Instruction for Mint<SignatureCheckCondition, Account> {}
+    impl Instruction for Mint<u32, Asset> {}
+    impl Instruction for Mint<u128, Asset> {}
+    impl Instruction for Mint<Fixed, Asset> {}
+    impl Instruction for Mint<u32, Trigger<TriggeringFilterBox>> {}
+
+    impl Instruction for Burn<PublicKey, Account> {}
+    impl Instruction for Burn<u32, Asset> {}
+    impl Instruction for Burn<u128, Asset> {}
+    impl Instruction for Burn<Fixed, Asset> {}
+    impl Instruction for Burn<u32, Trigger<TriggeringFilterBox>> {}
+
+    impl Instruction for Transfer<Account, DomainId, Account> {}
+    impl Instruction for Transfer<Account, AssetDefinitionId, Account> {}
+    impl Instruction for Transfer<Asset, u32, Account> {}
+    impl Instruction for Transfer<Asset, u128, Account> {}
+    impl Instruction for Transfer<Asset, Fixed, Account> {}
```

Comment:
we can't transfer store assets?

Reply:
Interesting... Nobody has mentioned that before. Looks like we have no impls in core...

I'll open a ticket

Reply:
- Created #4137